keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java

```diff
@@ -29,15 +31,16 @@ public AdminRoot(KeycloakSession session) {
      */
     @Path("realms")
     public Object getRealmsAdmin(@Context final HttpRequest request, @Context HttpResponse response) {
-        if (request.getHttpMethod().equals(HttpMethod.OPTIONS)) {
+        if (HttpMethod.OPTIONS.equals(request.getHttpMethod())) {
             return new AdminCorsPreflightService(request);
         }
 
         AdminAuth auth = authenticateRealmAdminRequest(request.getHttpHeaders());
-        if (auth != null) {
-            logger.debug("authenticated admin access for: " + auth.getUser().getUsername());
+        if (auth == null) {
+            throw new CloudtrustRuntimeException("Can't get AdminAuth");
```

Comment:
Shouln't we throw some kind of "UnauthorizedException" (to have a 401, and not a 500)?

Reply:
I did not want to make functional changes: in previous version, a NullPointerException would have been thrown further (when calling auth.getToken())
But you're right, we can fix this with an UnauthorizedException which would make sense

Reply:
Throwing NotAuthorizedException (UnauthorizedException is deprecated)